app/controllers/admin/export_domain_allows_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Admin
+  class ExportDomainAllowsController < BaseController
+    include AdminExportControllerConcern
+
+    ROWS_PROCESSING_LIMIT = 20_000
+
+    def new
+      authorize :domain_allow, :create?
+    end
+
+    def export
+      authorize :instance, :index?
+      send_export_file
+    end
+
+    def import
+      authorize :domain_allow, :create?
+      @import = Admin::Import.new(import_params)
+      parse_import_data!(export_headers)
+
+      @data.take(ROWS_PROCESSING_LIMIT).each do |row|
+        domain = row['#domain'].strip
+        next if DomainAllow.allowed?(domain)
+
+        domain_allow = DomainAllow.new(domain: domain)
+        log_action :create, domain_allow if domain_allow.save
+      end
+      redirect_to admin_instances_path, notice: I18n.t('admin.domain_allows.created_msg')
+    end
```

Comment:
`ActionController::ParameterMissing` should probably be rescued so that an helpful error message is given when no file is uploaded.